pkg/resource/function/hooks.go

```diff
@@ -105,7 +111,11 @@ func (rm *resourceManager) customUpdateFunction(
 	case delta.DifferentAt("Spec.Code"):
 		err = rm.updateFunctionCode(ctx, desired, delta)
 		if err != nil {
-			return nil, err
+			if strings.Contains(err.Error(), "Provide a valid source image.") {
```

Comment:
Could you add a comment above this with an example of the error, so we have context around this condition